src/components/ModalPin.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import { t } from 'ttag';
+import { connect } from "react-redux";
+import $ from 'jquery';
+import PinInput from './PinInput';
+import hathorLib from '@hathor/wallet-lib';
+import PropTypes from "prop-types";
+
+
+const mapStateToProps = (state) => {
+  return {
+    wallet: state.wallet,
+  };
+};
+
+/**
+ * Component that shows a modal with a form to ask for the user PIN
+ * and when the PIN succeeds, it invokes the callback function
+ * @memberof Components
+ */
+export class ModalPin extends React.Component {
+  /**
+   * errorMessage {string} Message to be shown to the user in case of error in the form
+   */
+  state = {
+    errorMessage: '',
+  }
+
+  pin = '';
+
+  componentDidMount = () => {
+    $('#modalPin').modal('show');
+    $('#modalPin').on('hidden.bs.modal', (e) => {
+      // Mandatory cleanup by GlobalModal
+      this.props.onClose();
+
+      // If the correct PIN was inserted, call the `onSuccess` callback
+      if (this.pin) {
+        this.props.onSuccess({ pin: this.pin })
+      }
+    });
+
+    // Focus the PIN field on modal load
+    $('#modalPin').on('shown.bs.modal', (e) => {
+      this.refs.pinInput.refs.pin.focus();
+    });
+  }
+
+  componentWillUnmount = () => {
+    // Removing all event listeners
+    $('#modalPin').off();
+    $('#modalPin').modal('hide');
+  }
+
+  /**
+   * Method called after user clicks the 'Go' button.
+   * We validate the form and that the pin is correct, then call a method from props.
+   *
+   * @param {React.MouseEvent<HTMLElement>} e Event emitted when button is clicked
+   */
+  handlePin = async (e) => {
+    e.preventDefault();
+
+    // Invalid form, show error message and do nothing else
+    if (this.refs.formPin.checkValidity() === false) {
+      this.refs.formPin.classList.add('was-validated');
+      return;
+    }
+
+    // Check if the pin is correct
+    this.refs.formPin.classList.remove('was-validated');
+    const pin = this.refs.pinInput.refs.pin.value;
+
+    // Incorrect PIN, show error message and do nothing else
+    if (!hathorLib.wallet.isPinCorrect(pin)) {
+      this.setState({ errorMessage: t`Invalid PIN` })
+      return;
+    }
+
+    // Set the PIN on the instance variable and close the modal.
+    this.pin = pin;
+
+    // Necessary callbacks will be executed at the `onHidden` modal event
+    $('#modalPin').modal('hide');
+  }
+
+  render() {
+    const renderBody = () => <div>
+        {this.props.bodyTop}
+        <form ref="formPin" onSubmit={this.handlePin} noValidate>
+          <div className="form-group">
+            <PinInput ref="pinInput" handleChangePin={this.props.handleChangePin}/>
```

Comment:
What's this `this.props.handleChangePin`? I don't see it defined

Reply:
Added this prop to the documentation on [008a5a3](https://github.com/HathorNetwork/hathor-wallet/pull/392/commits/008a5a30e96ca114745a4706bf4c795f4d55bacd).

Reply:
I saw that it's not required. Do we get a warning or error if we have this input with `undefined` as the handle method?

Reply:
No, it works fine! The parameter is actually optional at the `PinInput` component as well, even though it has no explicit documentation about it yet.